### ISSUE TITLE
Does this work with Embroider `staticAddonTrees` enabled?

### ISSUE THREAD

**Issue by tansongyang** [typed-ember/glint#277]
I work on a TS Ember app. We're currently building with Embroider, but I'd like to start turning on the flags to enable code splitting.

Per the [official Embroider recommendations](https://github.com/embroider-build/embroider#options), I started by turning on `staticAddonTestSupportTrees`. That worked and all my tests still pass.

But then I turned on `staticAddonTrees` and I got flooded with errors. I tried isolating this by running one of my component tests that had failed. This is the error:

<img width="577" alt="image" src="https://user-images.githubusercontent.com/9488719/161629098-80b26028-5218-4721-b064-9236e61c4f34.png">

And this is the line that fails:

```
let ApplePayButton = (_class = class ApplePayButton extends _glint_environment_ember_loose_glimmer_component__WEBPACK_IMPORTED_MODULE_9__["default"] {
```

So I tried changing my `Component` import back to `@glimmer/component` and now the error goes away. So I think this points to an issue with Glint.

Has Glint been tested with Embroider `staticAddonTrees` enabled? Is this supposed to be working? Or is it on the roadmap?

**Comment by jamescdavis** [comment]
The next minor version (0.8.0) of Glint will use native imports instead of the custom Glint imports. This release is expected very soon.

**Comment by tansongyang** [comment]
Awesome, thanks!

**Comment by ef4** [comment]
You can workaround this by manually registering the needed things with the runtime loader:

```js
// app.js

import * as GlimmerComponent from '@glimmer/component';
window.define('@glimmer/component', function () {
  return GlimmerComponent;
});

import * as Modifier from 'ember-modifier';
window.define('ember-modifier', function () {
  return Modifier;
});
```

**Comment by tansongyang** [comment]
@ef4 Thanks for the workaround! Out of curiosity, where would I have looked to find that documented? Or is this a dirty hack that we're trying not to disseminate?

**Comment by dfreeman** [comment]
The things we're doing in Glint with these reexports are absolutely dirty hacks, so while it may be that other cases exist in the wild where manually calling `define` works around issues with Embroider, I would hope they aren't _too_ common 😅

**Comment by tansongyang** [comment]
Haha, sounds good. Thanks for the help everyone! I'll leave it to you all whether to close this issue now or leave it open until an official fix ships.

**Comment by dfreeman** [comment]
As of Glint 0.7.4 native import paths are available, and as of version 0.8.0, Glint no longer exposes any runtime imports itself, so this should no longer be a problem! Check out [the migration guide](https://typed-ember.gitbook.io/glint/using-glint/migrating#glint-0.7.x-to-0.8.0) for further details.
